rust/src/index/vector.rs

```diff
@@ -316,10 +316,11 @@ pub(crate) async fn build_vector_index(
 }
 
 /// Open the Vector index on dataset, specified by the `uuid`.
-pub(crate) async fn open_index<'a>(
-    dataset: &'a Dataset,
+pub(crate) async fn open_index(
```

Comment:
Should the docs be updated for the new index?

Reply:
This index is not ready for public usage yet. After this issue, we still need some work to profiling and optimizations.